iib/workers/tasks/build_merge_index_image.py

```diff
@@ -308,33 +309,38 @@ def is_bundle_version_valid(bundle_path, valid_ocp_version):
     :return: a boolean indicating if the bundle_path satisfies the index ocp_version
     :rtype: bool
 
-           |  "v4.5"   |   "=v4.6"    | "v4.5-v4.7" | "v4.5,v4.6"  | "v4.6,v4.5"
+           |  "v4.5"   |   "=v4.6"    | "v4.5-v4.7"  | "v4.5,v4.6" | "v4.6,v4.5"
     -------------------------------------------------------------------------------
-    v4.5   | included  | NOT included |  included   |  included    |  NOT included
+    v4.5   | included  | NOT included |   included   |  included   |  NOT included
     -------------------------------------------------------------------------------
-    v4.6   | included  |   included   |  included   |  included    |  included
+    v4.6   | included  |   included   |   included   |  included   |  included
     -------------------------------------------------------------------------------
-    v4.7   | included  | NOT included |  included   |  included    |  included
+    v4.7   | included  | NOT included |   included   |  included   |  included
+    -------------------------------------------------------------------------------
+    v4.8   | included  | NOT included | NOT included |  included   |  included
     """
     try:
-        float_valid_ocp_version = float(valid_ocp_version.replace('v', ''))
+        ocp_version = valid_ocp_version.replace('v', '')
     except ValueError:
         raise IIBError(f'Invalid OCP version, "{valid_ocp_version}", specified in Index Image')
     try:
         bundle_version_label = get_image_label(bundle_path, 'com.redhat.openshift.versions')
         bundle_version = bundle_version_label.replace('v', '')
+        log.debug(f'Bundle path {bundle_path} version {bundle_version}, VALID version {valid_ocp_version}')
         if bundle_version_label.startswith('='):
-            if float(bundle_version.strip('=')) == float_valid_ocp_version:
+            if bundle_version.strip('=') == ocp_version:
                 return True
         elif '-' in bundle_version_label:
-            min_version, max_version = [float(version) for version in bundle_version.split('-')]
-            if min_version <= float_valid_ocp_version <= max_version:
+            min_version, max_version = [version for version in bundle_version.split('-')]
+
+            if sort_floats(list(min_version + ocp_version)) == list(min_version + ocp_version) and \
```

Comment:
The usage of lists seems unnatural here. Consider this. Instead of using `sort_floats`, introduce a new function, e.g. `to_version_tuple`, which does the conversion that is currently embedded in `sort_floats`, e.g. `'v4.10'` -> `(4, 10)`. Then, you can just compare the tuples as you'd compare any other value. `sort_floats` wouldn't be needed.

Reply:
what do you think about this? https://codereview.stackexchange.com/questions/217992/compare-two-version-numbers-in-the-form-major-minor-patch-build-to-see-if-one-is

Reply:
Ooh I like that! `LooseVersion` should do the trick for us.

```python
>>> from distutils.version import LooseVersion
>>> LooseVersion('4.10') > LooseVersion('4.9')
True
>>> LooseVersion('4.1') > LooseVersion('4.9')
False
>>> LooseVersion('4.10') > LooseVersion('4.101')
False
>>> LooseVersion('4.10') > LooseVersion('4.09')
True
```

Reply:
IT seems that this already deprecated. We can use this:
>>> from packaging import version
>>> version.parse("2.3.1") < version.parse("10.1.2")
True
>>> version.parse("1.3.a4") < version.parse("10.1.2")
True
>>> isinstance(version.parse("1.3.a4"), version.Version)
True
>>> isinstance(version.parse("1.3.xy123"), version.LegacyVersion)
True
>>> version.Version("1.3.xy123")
Traceback (most recent call last):
...
packaging.version.InvalidVersion: Invalid version: '1.3.xy123'